### PR TITLE
GPL-717 Use filtered positive fields

### LINKED ISSUE #242: GPL-717 Simplify positives report generation
**User story**
If we makes changes to filter positives earlier in the process in order to populate the Dart database, it should be possible to flag the filtered positives at that point in the MLWH and then just use that data to generate the positives report.

**Who are the primary contacts for this story**
PAM? Ops?

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] Change the STAGE_MATCH_POSITIVE and POSITIVE_SAMPLES_MONGODB_FILTER to use the new filtered_positive field: a sample is fit to pick if this is True
- [x] Dig around for any more duplications of the fit to pick rules, or RESULT=Positive assumptions, and remove these

**Dependencies**
This story is blocked by the following dependencies:
- https://github.com/sanger/crawler/issues/167

**Additional context**
Expecting to add some kind of flag for 'filtered positive' to the MLWH lighthouse_sample table. Should be able to select on those rows.
Assuming the positives report is even needed any more. In which case decommission the functionality.

### PR DESCRIPTION
**Pull request by pjvv**
Closes #242 

Changes proposed in this pull request:

* use the filtered_positive field when querying for positive samples
* removed old checks
* removed some typing warnings

### CODE REVIEW

**Review by emrojo — APPROVED** [review]

File: lighthouse/helpers/mysql_db.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import sqlalchemy  # type: ignore
-from sqlalchemy import MetaData  # type: ignore
-from sqlalchemy.engine.base import Engine  # type: ignore
+import sqlalchemy
```
Comment (andrewsparkes) [inline lighthouse/helpers/mysql_db.py:1]: When do or don't you need that type: ignore comment? Not clear to me...

File: tests/data/fixture_data.py [hunk 2]
```diff
@@ -163,50 +164,45 @@
         FIELD_RNA_ID: "rna_1",
         FIELD_LH_SAMPLE_UUID: "8e595a92-6798-4c93-8dc8-44f3ffb8bed3",
         FIELD_DATE_TESTED: "2020-05-10 07:30:00 UTC",
+        FIELD_FILTERED_POSITIVE: False,
     },
-    {  #  positive, with low Ct values
+    {  # positive, with low Ct values
```
Comment (andrewsparkes) [inline tests/data/fixture_data.py:169]: Are these tests with CT values relevant now? Comment not right at least, as you've removed the CQ values

File: tests/data/fixture_data.py [hunk 3]
```diff
@@ -344,6 +306,7 @@
         FIELD_COG_BARCODE: "hij",
         FIELD_ROOT_SAMPLE_ID: "MCM010",
         FIELD_DATE_TESTED: DATE_TESTED_NOW,
+        FIELD_FILTERED_POSITIVE: True,
     },
 ]
```
Comment (andrewsparkes) [inline tests/data/fixture_data.py:312]: Is there a test for filtered positive = null? just in case it doesn't get set for whatever reason.

**Review by andrewsparkes — APPROVED** [review]

File: lighthouse/helpers/mysql_db.py (hunk 1 above)
Comment (pjvv) [reply lighthouse/helpers/mysql_db.py:1]: I've added a `mypy.ini` config file which allows us to configure mypy and ignore packages which do not yet have [stub files](https://mypy.readthedocs.io/en/stable/stubs.html) in typeshep. There are edge cases where it will be easier to use `type: ignore` to tell mypy to ignore that line but it's usually worth the effort to resolve the type issues early on.

File: tests/data/fixture_data.py (hunk 2 above)
Comment (pjvv) [reply tests/data/fixture_data.py:169]: True, will have a look, thanks.

File: tests/data/fixture_data.py (hunk 3 above)
Comment (pjvv) [reply tests/data/fixture_data.py:312]: 🤔 good point, what are we expecting? Probably that that sample is deemed `filtered_positive = False`?

File: lighthouse/constants.py [hunk 4]
```diff
@@ -36,80 +46,52 @@
...
-                    {FIELD_CH2_CQ: {"$lte": CT_VALUE_LIMIT}},
-                    {FIELD_CH3_CQ: {"$lte": CT_VALUE_LIMIT}},
-                ],
-            },
-        ],
-        # 4. We are only interested in documents which have a valid date
+        # 1. We are only interested filtered positive samples
+        FIELD_FILTERED_POSITIVE: True,
+        # 2. We are only interested in documents which have a valid date
         FIELD_DATE_TESTED: {"$exists": True, "$nin": [None, ""]},
```
[the remark this answers is not in the document]
Comment (pjvv) [reply lighthouse/constants.py:63]: Nope, "Date Tested" is a required field.
